src/controllers/FavoritesController.ts

```diff
@@ -0,0 +1,200 @@
+                user.favorites?.splice(index, 1);
+                await user.save();
+            }
+        }
+
+        console.log("Deleted Favorite: ", favorite);
+        res.status(204).send();
+    } catch (error: any) {
+        console.error(`Could not delete the favorite with id ${req.query.id}\n`, error);
+        res.status(404).json({ error: "Could not delete the favorite with the specified id. Id does not exist" });
+    }
+});
+
+const getFavorites = asyncHandler(async (req, res) => {
+    const document = await Favorite.find();
+    if (!document || document.length === 0) {
+        res.status(404);
+        throw new Error("No favorites found");
+    }
+    const size = Array.isArray(document) ? document.length : 1;
+    res.status(200).json({ size, data: document });
+
+});
+
+const getFavoritesByCategoryId = asyncHandler(async (req, res) => {
+    try {
+        const document = await Favorite.find({ "type": req.query.categoryId });
+        if (!document || document.length === 0) {
+            res.status(404).json({ data: null });
+        }
+        const size = Array.isArray(document) ? document.length : 1;
+        res.status(200).json({ size, data: document });
+    } catch (error: any) {
+        console.error(`Could not find the category with id ${req.query.categoryId}\n`, error);
+        res.status(404).json({ error: "Could not find the category with the specified id" });
+    }
+});
+
+const getFavoritesByCategoryName = asyncHandler(async (req, res) => {
+    try {
+        if (req.query.category) {
+            const category = await CategoryModel.findOne({ "name": capitalizeFirstLetter(String(req.query.category)) });
+
+            if (!category) {
+                res.status(404).json({ error: "Could not find the category with the specified name" });
+            } else {
+                const favorites = await Favorite.find({ "type": category._id });
+                const size = Array.isArray(favorites) ? favorites.length : 1;
+                res.status(200).json({ size, data: favorites });
+            }
+        }
+    } catch (error: any) {
+        console.error(`Could not find the favorite with the name ${req.query.category}\n`, error);
+        res.status(404).json({ error: "Could not find the favorite with the specified name" });
+    }
+});
+
+const getFavoritesById = asyncHandler(async (req, res) => {
+    try {
+        const document = await Favorite.findById(req.query.id);
+        console.log(document);
+        if (!document) {
+            res.status(404).json({ error: "Could not find the favorite with the specified id" });
+        }
+        res.status(200).json({ data: document });
+    } catch (error: any) {
+        console.error(`Could not find the favorite with id ${req.query.id}\n`, error);
+        res.status(404).json({ error: "Could not find the favorite with the specified id" });
+    }
+});
+
+const searchFavorite = asyncHandler(async (req, res, next) => {
```

Comment:
This Endpoint will be removed.
The getFavoritesByCategoryName will be combined with getFavorites.

If query Parameter is in the request, the Favorites with the specified Category will be returned.
Otherwise all Favorites are returned.